### PR TITLE
Improve env sanity check to be more robust

### PR DESCRIPTION
**Pull request by Louis-Dupont**
Objectif:
- Support any time of requirements
- Improve the logic of version testing (current implementation is really getting dirty)

Besides, I also:
- Divided it into a few files: Motivation: facilitate reading.
- Moved all the error logging inside the checking functions themselves, instead of carrying them through and logging everyhting at the end. Motivation: making the code more clear and simple.

---
Note: 
I started with `semantic-version` (https://python-semanticversion.readthedocs.io/en/latest/reference.html) and `packaging`, but actually found a better way only using only `pkg_resources`. It supports the PyPi requirements standard.

### CODE REVIEW

**Review by shaydeci — CHANGES_REQUESTED** [review]
Looks good and robust now!
Just a few comments inline.

File: src/super_gradients/sanity_check/env_sanity_check.py [hunk 1]
```diff
@@ -39,138 +58,65 @@ def get_requirements_path(requirements_file_name: str) -> Union[None, Path]:
...
-        elif "==" in requirement:
-            constraint = "=="
-        else:
-            continue
+        pro_requirements = f.read().splitlines()
+
+    return requirements + pro_requirements if use_pro_requirements else requirements
+
+
+def check_packages(test_name: str):
```
[the remark this answers is not in the document]
Comment (Louis-Dupont) [reply src/super_gradients/sanity_check/env_sanity_check.py:77]: Now its directly set in the functions

File: src/super_gradients/sanity_check/env_sanity_check.py [hunk 2]
```diff
@@ -1,20 +1,39 @@
...
+
+    :param test_name:   Name of the test being tested.
+    :param error_msg:   Message to format in appropriate format.
+    :return:            Formatted message
+    """
+    return f"\33[31mFailed to verify {test_name}: {error_msg}\33[0m"
 
 
-def get_requirements_path(requirements_file_name: str) -> Union[None, Path]:
+def check_os(test_name: str):
```
[the remark this answers is not in the document]
Comment (Louis-Dupont) [reply src/super_gradients/sanity_check/env_sanity_check.py:25]: We only want to log errors when package version is smaller. When higher, we only want to log as debug

**Review by shaydeci — APPROVED** [review]
LGTM

**Pull request merged**